.github/workflows/test.yml

```diff
@@ -30,3 +30,76 @@ jobs:
 
       - name: Make Test
         run: make test
+
+  test-kotlin-libraries:
+    name: Test Kotlin Libraries
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - api-level: 28
+            arch-ndk: x86_64
+            arch-cargo: x86_64-linux-android
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Enable KVM
+        run: |
+          echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
+          sudo udevadm control --reload-rules
+          sudo udevadm trigger --name-match=kvm
+
+      - name: Cargo toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.81.0 # Check the version used by Holochain
+          targets: ${{ matrix.target.arch-cargo }}
+
+      - name: Gradle cache
+        uses: gradle/actions/setup-gradle@v3
+        
+      - name: AVD cache
+        uses: actions/cache@v4
+        id: avd-cache
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-${{ matrix.target.api-level }}-${{ matrix.target.arch-ndk }}
+
+      - name: Cargo cache
+        uses: actions/cache@v4
+        id: cargo-cache
+        with:
+          path: |
+            ~/.cargo/*
+            ~/target/*
+          key: cargo-${{ matrix.target.arch-cargo }}
+
+      - uses: pnpm/action-setup@v4
+        name: Install pnpm
+        with:
+          version: 10
+
+      - name: Install cargo-ndk
+        run: |
+          cargo install cargo-ndk
+
+      - name: Build dependency rust library & kotlin bindings
+        run: |
+          pnpm run build:single-target:holochain-conductor-runtime-types-ffi ${{ matrix.target.arch-cargo }}
+      
+      - name: Create gradle.properties without secrets
+        run: |
+          touch libraries/holochain-service-client/gradle.properties
+          printf "android.enableJetifier=true\nandroid.useAndroidX=true" > libraries/holochain-service-client/gradle.properties
```

Comment:
Secrets are needed for publishing to Maven Central, hence why the file is excluded from the repo